src/Ledger/Utxo.lagda

```diff
@@ -320,14 +320,15 @@ isAdaOnlyᵇ v = toBool (policies v ≡ᵉ coinPolicies)
 \end{code}
 \begin{code}
 
-feesOK : PParams → Tx → UTxO → Bool
-feesOK pp tx utxo = minfee pp utxo tx ≤ᵇ txfee
-                  ∧ not (≟-∅ᵇ (txrdmrs ˢ))
-                  =>ᵇ ( allᵇ (λ (addr , _) → ¿ isVKeyAddr addr ¿) collateralRange
-                      ∧ isAdaOnlyᵇ bal
-                      ∧ (coin bal * 100) ≥ᵇ (txfee * pp .collateralPercentage)
-                      ∧ not (≟-∅ᵇ collateral)
-                      )
+feesOK : (pp : PParams) → paramsWellFormed pp → Tx → UTxO → Bool
+feesOK pp ppwf tx utxo =  (  minfee pp ppwf utxo tx ≤ᵇ txfee ∧ not (≟-∅ᵇ (txrdmrs ˢ))
+                             =>ᵇ  ( allᵇ (λ (addr , _) → ¿ isVKeyAddr addr ¿) collateralRange
+                                  ∧ isAdaOnlyᵇ bal
+                                  ∧ (coin bal * 100) ≥ᵇ (txfee * pp .collateralPercentage)
+                                  ∧ not (≟-∅ᵇ collateral)
+                                  )
+                          )
+                          ∧ scriptsCost pp ppwf utxo tx ≤ᵇ pp .maxRefScriptPerTx
```

Comment:
The LHS here shouldn't be the cost but the total size. It also doesn't make sense to be in `feesOK`.

Reply:
Okay, I've removed it from `feesOK` and added a premise for the UTXO rule instead.